fix: add Sentry error logging to PrismaAppRepository methods

Vansh5632 opened the pull request:
## What does this PR do 
- Add try-catch blocks around Prisma queries in findAppStore and seedApp methods
- Import captureException from @sentry/nextjs for error logging
- Maintain error propagation by re-throwing caught errors
- Fix TypeScript type issue (any -> object) in seedApp method
- Ensure consistent error handling pattern across repository methods

This resolves database connection failures not being logged to Sentry, making debugging database issues much easier and improving error monitoring.

- Fixes #24027  (GitHub issue number)
- Fixes CAL-6455 (Linear issue number - should be visible at the bottom of the GitHub issue description).

#### Video Demo (if applicable):

https://github.com/user-attachments/assets/d84eb747-5d1a-4a65-9a83-b91062ec389c



## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

Vansh5632 committed:
fix: add Sentry error logging to PrismaAppRepository methods

- Add try-catch blocks around Prisma queries in findAppStore and seedApp methods
- Import captureException from @sentry/nextjs for error logging
- Maintain error propagation by re-throwing caught errors
- Fix TypeScript type issue (any -> object) in seedApp method
- Ensure consistent error handling pattern across repository methods

This resolves database connection failures not being logged to Sentry,
making debugging database issues much easier and improving error monitoring.

youneshenniwrites commented on packages/lib/server/repository/PrismaAppRepository.ts:
```diff
@@ -1,26 +1,39 @@
-      throw new Error(`App ${dirName} not found`);
-    }
+      if (!appMetadata) {
+        throw new Error(`App ${dirName} not found`);
```
Why no Sentry `captureException` here ?

Vansh5632 committed:
fix: add Sentry logging for app metadata not found error

Add captureException for the manual error thrown when appMetadata
is not found to ensure consistent error monitoring across all
error paths in the PrismaAppRepository.

This ensures that invalid app directory names are also tracked
and logged to Sentry for better debugging and monitoring.
